bessctl/commands.py

```diff
@@ -694,11 +694,18 @@ def _do_start(cli, opts):
     except subprocess.CalledProcessError:
         try:
             cli.bess.connect()  # reconnect to the old instance, if any
-        except:
+        except cli.bess.APIError:
             pass
         raise cli.CommandError('Cannot start BESS daemon')
     else:
-        cli.bess.connect()
+        start = time.time()
+        while time.time() - start < 3:
+            try:
+                cli.bess.connect()
+                break
+            except cli.bess.APIError:
+                # bessd is on, but its gRPC server may be not yet. Retry.
+                time.sleep(0.1)
```

Comment:
We should probably test whether we're connected once we get out of the loop.  Other than that, this change itself seems fine.

More generally, I'm not sure but I think maybe this whole startup process should be reconstructed a bit.  I don't understand why we catch `subprocess.CalledProcessError` and try to connect to the old bessd - shouldn't failure to start a new one just be failure?

Reply:
Oh I thought I added it. Fixed.

Reconnecting to the old instance has hardly helped, since it would have already been killed by `bessd -k` with a 99.9% chance :) I removed it.